src/main/scala/modbat/mbt/Main.scala

```diff
@@ -28,13 +29,13 @@ object Main {
 	  if (!remainingArgs.hasNext) {
 	    Log.error(c.header)
 	    Log.error("Model class argument missing. Try --help.")
-	    return 1
+      throw new IllegalArgumentException(c.header)
 	  }
 	  modelClassName = remainingArgs.next
 	  if (remainingArgs.hasNext) {
 	    Log.error("Extra arguments starting at \"" + remainingArgs.next() +
 		      "\" are not supported.")
-	    return 1
+      throw new IllegalArgumentException(remainingArgs.next())
```

Comment:
This will not work, because `remainingArgs.next` advances the iterator by one position.
If you want to use that as part of the message, you have to store the result of `remainingArgs.next` in a temporary variable before using it for logging and as part of the exception message string.